aws/src/test/java/org/apache/iceberg/aws/s3/S3OutputStreamTest.java

```diff
@@ -133,6 +135,65 @@ public void testAbortMultipart() {
     }
   }
 
+  @Test(expected = IllegalStateException.class)
```

Comment:
Nit: can we do `AssertHelpers.assertThrows(<message>, IllegalStateException.class, () -> stream.write(1));` in these test cases instead of `expected` to check against exception message, to ensure it's the same exception being thrown as expected?

Reply:
yeah. this is more precise. updated the code to use `AssertHelpers.assertThrows`. only annoying thing is the unchecked exception, which make the code verbose